pkg/pool-weighted/test/foundry/LiquidityApproximationWeighted.t.sol

```diff
@@ -170,11 +170,35 @@ contract LiquidityApproximationWeightedTest is LiquidityApproximationTest {
         uint256 exactBptAmount,
         uint256 swapFeePercentage,
         uint256 weightDai
+    ) public {
+        exactBptAmount = bound(exactBptAmount, minAmount, maxAmount / 2 - 1);
+        swapFeePercentage = bound(swapFeePercentage, minSwapFeePercentage, maxSwapFeePercentage);
+        weightDai = bound(weightDai, 20e16, 80e16);
+
+        // Weights can introduce some differences in the swap fees calculated by the pool during unbalanced add/remove
+        // liquidity, so the error tolerance needs to be a bit higher than the default tolerance.
+        excessRoundingDelta = 1e16; // 0.1%
+        defectRoundingDelta = 0.001e16; // 0.001%
+
+        addLiquidityProportionalAndRemoveExactInWeights(exactBptAmount, swapFeePercentage, weightDai);
+    }
+
+    /// @dev Same as testAddLiquidityProportionalAndRemoveExactInWeights__Fuzz, with more tolerance (extreme case).
+    function testAddLiquidityProportionalAndRemoveExactInExtremeWeights__Fuzz(
+        uint256 exactBptAmount,
+        uint256 swapFeePercentage,
+        uint256 weightDai
     ) public {
         exactBptAmount = bound(exactBptAmount, minAmount, maxAmount / 2 - 1);
         swapFeePercentage = bound(swapFeePercentage, minSwapFeePercentage, maxSwapFeePercentage);
         weightDai = bound(weightDai, 1e16, 99e16);
 
+        // TODO: check why the difference grows so large in the extremes (high fees, high dai weight).
```

Comment:
As discussed, RemoveLiquiditySingleTokenExactIn charges the fee in token out, which amplifies the error when compared with the Swap, that charges the fee in token in.
Maybe this is not the whole explanation, but if it is, we could remove this TODO and add this comment here (and wherever we compare RemoveLiquiditySingleTokenExactIn with Swap).

Reply:
These tests will get one more pass. For now I'm just fixing them and taking some notes; the main point is to get the code merged for now.